pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -443,8 +443,19 @@ func containerChanged(container *v1.Container, containerStatus *kubecontainer.Co
 	return expectedHash, containerStatus.Hash, containerStatus.Hash != expectedHash
 }
 
-func shouldRestartOnFailure(pod *v1.Pod) bool {
-	return pod.Spec.RestartPolicy != v1.RestartPolicyNever
+func shouldRestartOnFailure(pod *v1.Pod, containerStatus *kubecontainer.ContainerStatus) bool {
+	switch pod.Spec.RestartPolicy {
+	case v1.RestartPolicyAlways:
+		return true
+	case v1.RestartPolicyOnFailure:
+		if containerStatus == nil {
```

Comment:
Why do we need to pass the container status into this function? Could we extract it from `pod`?

Reply:
Yes, it could be extracted from `pod`, then we need pass container name. Currently there is container status in hand, so I pass container status.